src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/ReclassifyCallbackService.java

```diff
@@ -0,0 +1,119 @@
+                    + "Exception record ID: {}. Jurisdiction: {}. User ID: {}",
+                exceptionRecordId,
+                jurisdiction,
+                userId
+            );
+
+            return new ProcessResult(updateClassification(exceptionRecordDetails.getData()));
+        } else {
+            // Logging validation failure as error, because the caseworker should have no way of causing it.
+            // If it happens, the team needs to be alerted.
+            log.error(
+                "Validation failed for exception record reclassification. "
+                + "Exception record ID: {}. Jurisdiction: {}. User ID: {}. Errors: {}",
+                exceptionRecordId,
+                jurisdiction,
+                userId,
+                String.join(", ", errors)
+            );
+
+            return new ProcessResult(emptyList(), errors);
+        }
+    }
+
+    private Map<String, Object> updateClassification(Map<String, Object> exceptionRecordData) {
+        Map<String, Object> updatedFields = Maps.newHashMap(exceptionRecordData);
+        updatedFields.put(JOURNEY_CLASSIFICATION, SUPPLEMENTARY_EVIDENCE_WITH_OCR.name());
+        return updatedFields;
+    }
+
+    private List<String> validateCallbackRequest(CaseDetails exceptionRecordDetails) {
+        List<Optional<String>> validationErrors = asList(
+            validateState(exceptionRecordDetails),
+            validateClassification(exceptionRecordDetails)
+        );
+
+        return validationErrors
+            .stream()
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(toList());
+    }
+
+    private Optional<String> validateState(CaseDetails exceptionRecordDetails) {
+        String currentState = exceptionRecordDetails.getState();
+
+        if (!Objects.equals(currentState, REQUIRED_INITIAL_STATE)) {
+            return Optional.of(
+                String.format(
+                    "Reclassification is only allowed for '%s' state. This Exception record's current state is '%s'.",
+                    REQUIRED_INITIAL_STATE,
+                    currentState
+                )
+            );
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private Optional<String> validateClassification(CaseDetails exceptionRecordDetails) {
+        String classification = exceptionRecordDetails.getData().get(JOURNEY_CLASSIFICATION).toString();
+        if (!Objects.equals(classification, REQUIRED_INITIAL_CLASSIFICATION)) {
+            return Optional.of(
+                String.format(
+                    "This exception record's journey classification is '%s'. "
+                        + "Reclassification is only possible from '%s' classification.",
+                    classification,
+                    REQUIRED_INITIAL_CLASSIFICATION
+                )
+            );
+        } else {
+            return Optional.empty();
+        }
+    }
```

Comment:
Kind of beats the purpose of optional response in conjunction with if/else conditions. would it look ugly if we try something functionally here?

```java
return Optional.ofNullable(...)
    .map(cast to string if necessary)
    .filter(v -> !v.equals(...))
    .map(format validation fail message);
```

Reply:
I agree that once you've got an Optional, especially in Java 11, using if-else may make little sense, but this code creates an Optional instead of dealing with an existing one. While it's possible to replace pretty much every kind of logic with a chain of transformations, I think it would be a bit artificial here, given how simple this case is.